File: deploy-platform/backend/app/schemas/patrol.py
from datetime import datetime
from pydantic import BaseModel


class NodePatrolResult(BaseModel):
    node_id: int
    host_ip: str
    service_name: str
    status: str
    detail: str
    checked_at: datetime | None = None


class PatrolRunRequest(BaseModel):
    environment_id: int
    service_ids: list[int] | None = None


class PatrolRunResponse(BaseModel):
    environment_id: int
    total_nodes: int
    healthy_nodes: int
    unhealthy_nodes: int
    results: list[NodePatrolResult]
    checked_at: datetime

File: deploy-platform/frontend/src/api/patrol.ts
import client from './client'

export interface NodePatrolResult {
  node_id: number
  host_ip: string
  service_name: string
  status: string
  detail: string
  checked_at: string | null
}

export interface PatrolRunResponse {
  environment_id: number
  total_nodes: number
  healthy_nodes: number
  unhealthy_nodes: number
  results: NodePatrolResult[]
  checked_at: string
}

export function runPatrol(environmentId: number, serviceIds?: number[]): Promise<PatrolRunResponse> {
  return client.post('/patrol/run', { environment_id: environmentId, service_ids: serviceIds ?? null }).then((r) => r.data)
}
